src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/UriSearchValue.cs

```diff
@@ -53,6 +54,23 @@ public void AcceptVisitor(ISearchValueVisitor visitor)
             visitor.Visit(this);
         }
 
+        public bool Equals([AllowNull] ISearchValue other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            var uriSearchValueOther = other as UriSearchValue;
+
+            if (uriSearchValueOther == null)
+            {
+                return false;
+            }
```

Comment:
This null check and cast is used so often I feel it should be a helper method.

Reply:
The equals method can be slightly different on the last line, so for simplicity I think I will leave it as is.